src/libUtils/DataConversion.cpp

```diff
@@ -94,3 +94,10 @@ const std::string DataConversion::CharArrayToString(
 
   return ret;
 }
+
+uint16_t DataConversion::charArrTo16Bits(
+    const std::vector<unsigned char>& hex_arr) {
+  uint32_t lsb = hex_arr.size() - 1;
```

Comment:
can check whether is hex_arr empty or not. Else they may be an underflow

Reply:
Fixed